pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -291,3 +291,132 @@ func NewNamedMapFromBuildPlan(build *model.Build) (artifact.NamedMap, error) {
 
 	return res, nil
 }
+
+// BuildtimeArtifacts iterates through all artifacts in a given build and
+// adds the artifact's dependencies to a map. This is different from the
+// runtime dependency calculation as it includes ALL of the input artifacts of the
+// step that generated each artifact. This is useful for when we are using the
+// BuildLogStreamer as it operates on the older recipeID and will include
+// more artifacts than what we originally calculated in the runtime closure.
+func BuildtimeArtifacts(build *model.Build) (artifact.Map, error) {
+	result := make(artifact.Map)
+	lookup := make(map[strfmt.UUID]interface{})
+
+	for _, artifact := range build.Artifacts {
+		lookup[artifact.NodeID] = artifact
+	}
+	for _, step := range build.Steps {
+		lookup[step.StepID] = step
+	}
+	for _, source := range build.Sources {
+		lookup[source.NodeID] = source
+	}
+
+	for _, a := range build.Artifacts {
+		// Since we are using the BuildLogStreamer, we need to add all of the
+		// artifacts that have been submitted to be built except for the artifacts
+		// of the builder mime type as the BuildLogStreamer does not send events
+		// for builder artifacts.
+		if a.MimeType == model.XActiveStateBuilderMimeType {
+			continue
+		}
+
+		_, ok := result[strfmt.UUID(a.NodeID)]
+		if !ok && a.Status != model.ArtifactNotSubmitted {
+			deps := make(map[strfmt.UUID]struct{})
+			for _, depID := range a.RuntimeDependencies {
+				artifact, ok := lookup[depID].(*model.Artifact)
+				if ok {
+					if artifact.MimeType == model.XActiveStateBuilderMimeType {
+						logging.Debug("Skipping active state builder artifact: %s", depID)
+						continue
+					}
+				}
+
+				deps[depID] = struct{}{}
+				recursiveDeps, err := generateBuildtimeDependencies(depID, lookup, deps)
+				if err != nil {
+					return nil, errs.Wrap(err, "Could not resolve runtime dependencies for artifact: %s", depID)
+				}
+				for id := range recursiveDeps {
```

Comment:
Does this work when `generateBuildTimeDepenencies` returns `nil, nil`?

Reply:
By work do you mean not error? If so, then yes it will work.

Reply:
Yeah, not sure how go handles iterating over a `nil`.. I would've expected a panic.

Reply:
Let me test it in the go playground just to be safe :P

Reply:
Yeah, shouldn't panic: https://go.dev/play/p/jkOK0owZEcI